### PR TITLE
feat: Add comprehensive tests for natural language search with BM25 scoring

### PR DESCRIPTION
**Pull request by cozy-corner**
## Summary
- Add 13 comprehensive test cases for `searchFishByNaturalLanguage` function
- Fix empty query handling and special character sanitization
- Include BM25 score in search results for relevance tracking

### Bug Fixes
- **Empty query handling**: Return empty array instead of causing SQL error
- **Special character sanitization**: Remove characters that cause FTS5 syntax errors (`&`, `@`, `#`, etc.)
- **Score inclusion**: Add BM25 score to `FishWithMatch` type and results

### Test Coverage
- Basic natural language queries with relevant terms
- BM25 score threshold filtering and ranking validation
- Japanese language query handling
- Multi-word and single-word query processing
- Empty query and special character edge cases
- Limit parameter validation
- Score inclusion and ordering verification

## Test Plan
- [x] All existing tests continue to pass (46/46)
- [x] New natural language search tests added (13 test cases)
- [x] Pre-commit hooks pass (lint, format, typecheck, tests)
- [x] CI-equivalent checks pass

## Files Changed
- `src/services/search-service.ts`: Add query sanitization, empty query handling, score field
- `src/services/__tests__/search-service.test.ts`: Add comprehensive test suite for natural language search
- `TODO.md`: Update progress tracking

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Added configurable BM25 score threshold to natural language fish search, allowing users to adjust search result relevance.
  - Search results now include a relevance score for each match.
- **Bug Fixes**
  - Improved handling of empty queries and special characters in search to prevent errors.
- **Documentation**
  - Added documentation explaining the BM25 threshold parameter and its configuration.
- **Tests**
  - Introduced comprehensive tests for natural language search, covering query handling, result ranking, and threshold filtering.
- **Chores**
  - Added a debug script for testing and inspecting search results.

### CODE REVIEW

File: src/services/search-service.ts [hunk 1]
```diff
@@ -308,9 +313,26 @@ export class SearchService {
...
     // BM25スコアの閾値（負の値が高スコア）
     const SCORE_THRESHOLD = -10.0;
 
+    // Sanitize query for FTS5
+    // Remove special characters that can cause syntax errors
+    const sanitizedQuery = query
+      .replace(/[&@#<>(){}[\]"']/g, ' ')
+      .replace(/\s+/g, ' ')
+      .trim();
+
```
[the remark this answers is not in the document]
Comment (cozy-corner) [reply src/services/search-service.ts]: Fixed in commit [bda9327](https://github.com/cozy-corner/fish-mcp-server/commit/bda9327). Removed complex regex sanitization and replaced with minimal handling of FTS5 operators (&@#) that actually cause syntax errors.

[the remark this answers is not in the document]
Comment (cozy-corner) [reply src/services/search-service.ts]: Fixed in commit 59ff9a7. SQLite FTS5 intentionally returns negative BM25 values (multiplied by -1) for convenient ASC sorting. The correct operator is `<` for filtering negative scores, where more negative values indicate higher relevance.

[the remark this answers is not in the document]
Comment (cozy-corner) [reply src/services/search-service.ts]: Fixed in commit [61bb31e](https://github.com/cozy-corner/fish-mcp-server/commit/61bb31e). Updated comment to correctly reflect SQLite FTS5 BM25 behavior: positive values where smaller values indicate higher relevance.
